docs/NNNN-hashed-n-tuple-storage-layout.md

```diff
@@ -0,0 +1,242 @@
+# OCFL Community Extension NNNN: Hashed Truncated N-tuple Storage Layout
+
+  * **Extension Name:** NNNN-hashed-n-tuple-storage-layout
+  * **Authors:** Peter Winckles
+  * **Minimum OCFL Version:** 1.0
+  * **Obsoletes:** n/a
+  * **Obsoleted by:** n/a
+
+## Overview
+
+This storage root extension describes how to safely map OCFL object identifiers of any length, containing any characters to OCFL object root directories with the primary goals of ensuring portability and filesystem performance at the cost of directory name transparency.
+
+Using this extension, OCFL object identifiers are hashed and encoded as hex strings. These digests are then divided into _N_ n-tuple segments, which are used to create nested paths under the OCFL storage root.
+
+This approach allows OCFL object identifiers of any composition to be evenly distributed across the storage hierarchy. The maximum number of files under any given directory is controlled by the number of characters in each n-tuple, and the tree depth is controlled by the number of n-tuple segments each digest is divided into. Additionally, it obviates the need to handle special characters in OCFL object identifiers because the mapped directory names will only ever contain the characters `0-9a-f`.
+
+However, this comes at the cost of not being able to identify the OCFL object identifier of an object simply by browsing the OCFL storage hierarchy. The ID of an object may only be found within its `inventory.json`.
+
+## Parameters
+
+### Summary
+
+* **Name:** `digestAlgorithm`
+  * **Description:** The digest algorithm to apply on the OCFL object identifier; MUST be an algorithm that is allowed in the OCFL fixity block
+  * **Type:** string
+  * **Range:** 1,4095
```

Comment:
I don't think it really matters but 4095 seems a very long `digestAlgorithm`. However, I'm not sure what arbitrarily shorter length would be better

Reply:
Yes, I'm more than happy to change this once parameter constraint specification is clarified